libs/mpi/docs/index.rst

```diff
@@ -0,0 +1,18 @@
+..
+    Copyright (c) 2019 The STE||AR-Group
+
+    SPDX-License-Identifier: BSL-1.0
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+.. _libs_mpi:
+
+===
+mpi
+===
+
+TODO: High-level description of the library.
```

Comment:
@biddisco this is a good place to write documentation on using MPI explicitly in HPX.